api/api.go

```diff
@@ -24,12 +26,24 @@ import (
 	"github.com/prometheus/alertmanager/provider"
 	"github.com/prometheus/alertmanager/silence"
 	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/common/route"
 
 	"github.com/go-kit/kit/log"
 )
 
+var (
+	requestsInFlight = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "alertmanager_http_get_requests_in_flight",
+		Help: "Current number of HTTP GET requests being processed.",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(requestsInFlight)
```

Comment:
I am not a big fan of the global metric registry. It might be less verbose but also implicit and more difficult when it comes to vendoring. I am trying the steer this project away from it. What do you think in this regard? Would passing a `Registerer` to `API.New()` also be an option?

Reply:
I'm not a fan of the global registry, either. I just wanted to stay consistent with the way it is and then have a separate refactoring to make it local.

But now I have seen what you did in #1741, which is already merged. Thus, a incremental migration to local Registerers is already ongoing. Happy to follow that path here.